partitions.go

```diff
@@ -111,6 +113,19 @@ func (p *partitions) updateLocalPartitions(local map[int]bool) {
 	}
 }
 
+func (p *partitions) deDupe(nodes []string) []string {
+	found := map[string]bool{}
+	dedupedNodes := make([]string, len(nodes))
```

Comment:
this should be `make([]string, 0, len(nodes))` or you're going to have extra empty values at the end